impl/src/fmt/display.rs

```diff
@@ -224,70 +242,134 @@ impl<'a> Expansion<'a> {
+                    if let Some((expr, trait_ident)) = fmt.transparent_call() {
+                        if self.shared_attr.is_some() {
+                            let placeholder =
+                                trait_name_to_default_placeholder_literal(&trait_ident);
+
+                            quote! { derive_more::core::format_args!(#placeholder, #expr) }
+                        } else {
+                            quote! {
+                                derive_more::core::fmt::#trait_ident::fmt(&(#expr), __derive_more_f)
+                            }
+                        }
+                    } else if self.shared_attr.is_some() {
+                        quote! { derive_more::core::format_args!(#fmt) }
+                    } else {
+                        quote! { derive_more::core::write!(__derive_more_f, #fmt) }
+                    }
+                }
+                None if self.fields.is_empty() => {
+                    let ident_str = self.ident.to_string();
+
+                    if self.shared_attr.is_some() {
+                        quote! { #ident_str }
+                    } else {
+                        quote! { derive_more::core::write!(__derive_more_f, #ident_str) }
+                    }
+                }
+                None if self.fields.len() == 1 => {
+                    let field = self
+                        .fields
+                        .iter()
+                        .next()
+                        .unwrap_or_else(|| unreachable!("count() == 1"));
+                    let ident =
+                        field.ident.clone().unwrap_or_else(|| format_ident!("_0"));
+                    let trait_ident = self.trait_ident;
+
+                    if self.shared_attr.is_some() {
+                        let placeholder =
+                            trait_name_to_default_placeholder_literal(trait_ident);
+
+                        quote! { derive_more::core::format_args!(#placeholder, #ident) }
+                    } else {
+                        quote! {
+                            derive_more::core::fmt::#trait_ident::fmt(#ident, __derive_more_f)
+                        }
+                    }
+                }
+                _ => {
+                    return Err(syn::Error::new(
+                        self.fields.span(),
+                        format!(
+                            "struct or enum variant with more than 1 field must have \
                      `#[{}(\"...\", ...)]` attribute",
-                    trait_name_to_attribute_name(self.trait_ident),
-                ),
-            )),
+                            trait_name_to_attribute_name(self.trait_ident),
+                        ),
+                    ))
+                }
+            };
         }
+
+        if let Some(shared_fmt) = &self.shared_attr {
+            let shared_body = if let Some((shared_expr, shared_trait_ident)) =
+                shared_fmt.transparent_call()
```

Comment:
@JelteF ahh, yes... that's why I wanted to extend tests suite a little bit more. Good catch, again.

This issue is fixable by additionally referencing the `format_args!()` expression. Cove

> I think it would be best if we just ignored the whole shared_attr logic if the shared attr (both here in and in the big code block above). i.e. if the shared attr is `"{_variant}"` then we consider shared_attr to be `None`.

Yes, I was thinking about it too, but left "as is" as a paved way for future supporting of `{_variant}` with modifiers.

Reply:
However, thinking about it more... one thing does not exclude another one.